### PR TITLE
Spark Max Fixes

### PR DESCRIPTION
**Pull request by ManiacMoustache**
Spark maxs are fiddly. No idea how we fixed them. We need to figure these things out.

### CODE REVIEW

File: robotMap.py [hunk 1]
```diff
@@ -35,13 +35,15 @@ def controllerInput(self):
...
         #Mechanism controller inputs
         self.mechLeft = self.mech.getRawAxis(XboxController.Axis.kLeftY)
         self.mechRight = self.mech.getRawAxis(XboxController.Axis.kRightY)
         self.mechLeftHoriz = self.mech.getRawAxis(XboxController.Axis.kLeftX)
         self.mechRightHoriz = self.mech.getRawAxis(XboxController.Axis.kRightX)
         self.mechRightTrig = self.mech.getRawAxis(XboxController.Axis.kRightTrigger)
         self.mechLeftTrig = self.mech.getRawAxis(XboxController.Axis.kLeftTrigger)
+        self.mechDPad = self.mech.getPOV()
 
     def getDriveController(self):
```
Comment (DrKittyKat) [inline robotMap.py]: None of these have any documentation and I'm not quite sure what they do. I would appreciate some description

File: components/lifter.py [hunk 2]
```diff
@@ -6,12 +6,14 @@ class Lifter:
...
         self.upSpeed = 0
         self.winchMotor = self.motors_lifter["winchMotor"]
-        self.winchFollower = self.motors_lifter["winchFollower"]
 
         self.logger.info("Lifter Component Created")
     
-    def setSpeed(self, speed):
-        self.upSpeed = speed
-    
+    def setRaise(self):
```
Comment (DrKittyKat) [inline components/lifter.py]: comments might be nice

File: robot.py [hunk 3]
```diff
@@ -84,7 +84,11 @@ def teleopPeriodic(self):
 
         self.driveTrain.setArcade(driveLeft, driveRight)
 
-        # Scoprion Code
+        if self.xboxMap.getMechDPad() == 0:
+            self.lifter.setRaise()
+        else:
+            self.lifter.stop()
+        # Scorpion Code
```
Comment (DrKittyKat) [inline robot.py]: Which part of this is specific for scorpion?

**Comment by ChrisHirsch** [comment]
This PR seems to crash the sim:
  File ".\robot.py", line 121, in instantiateSubsystemGroup
    items = {key:factory(descp) for (key, descp) in config.getGroupDict(subsystem, groupName).items()}
  File ".\robot.py", line 121, in <dictcomp>
    items = {key:factory(descp) for (key, descp) in config.getGroupDict(subsystem, groupName).items()}
  File "Y:\Users\chirsch\src\raptacon\Robot-2020\utils\motorHelper.py", line 38, in createMotor
    setTalonFXCurrentLimits(motor, motorDescp)
  File "Y:\Users\chirsch\src\raptacon\Robot-2020\utils\motorHelper.py", line 80, in setTalonFXCurrentLimits
    triggerThresholdCurrent = currentLimits['triggerThresholdCurrent']
KeyError: 'triggerThresholdCurrent'

File: components/lifter.py (hunk 2 above)
Comment (ChrisHirsch) [reply components/lifter.py]: Afgreed...merging for now...but any future functions please set comments on them.

**Comment by ChrisHirsch** [comment]
> This PR seems to crash the sim:
> File ".\robot.py", line 121, in instantiateSubsystemGroup
> items = {key:factory(descp) for (key, descp) in config.getGroupDict(subsystem, groupName).items()}
> File ".\robot.py", line 121, in
> items = {key:factory(descp) for (key, descp) in config.getGroupDict(subsystem, groupName).items()}
> File "Y:\Users\chirsch\src\raptacon\Robot-2020\utils\motorHelper.py", line 38, in createMotor
> setTalonFXCurrentLimits(motor, motorDescp)
> File "Y:\Users\chirsch\src\raptacon\Robot-2020\utils\motorHelper.py", line 80, in setTalonFXCurrentLimits
> triggerThresholdCurrent = currentLimits['triggerThresholdCurrent']
> KeyError: 'triggerThresholdCurrent'

This actually wasn't a probem with this PR. It was already in Master and has been fixed

**Pull request merged**